src/jenkins/ComponentBuildStatus.groovy

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package jenkins
+
+import groovy.json.JsonOutput
+import utils.OpenSearchMetricsQuery
+
+class ComponentBuildStatus {
+    String metricsUrl
+    String awsAccessKey
+    String awsSecretKey
+    String awsSessionToken
+    String indexName
+    String product 
+    String version
+    String distributionBuildNumber
+    String buildStartTimeFrom
+    String buildStartTimeTo
+    def script
+
+    ComponentBuildStatus(String metricsUrl, String awsAccessKey, String awsSecretKey, String awsSessionToken, String indexName, String product, String version, String distributionBuildNumber, String buildStartTimeFrom, String buildStartTimeTo, def script) {
+        this.metricsUrl = metricsUrl
+        this.awsAccessKey = awsAccessKey
+        this.awsSecretKey = awsSecretKey
+        this.awsSessionToken = awsSessionToken
+        this.indexName = indexName
+        this.product = product
+        this.version = version
+        this.distributionBuildNumber = distributionBuildNumber
+        this.buildStartTimeFrom = buildStartTimeFrom
+        this.buildStartTimeTo = buildStartTimeTo
+        this.script = script
+    }
+
+    def getQuery(String componentBuildResult) {
```

Comment:
Where are these funcs or this lib file being called?

Reply:
getQuery is called below in getComponents and getComponents is called here https://github.com/opensearch-project/opensearch-build-libraries/pull/515/files#diff-6c9cab58a62fc361bc8e4698777911100f056e59015d9e6e7b3ca430f12cd6e5R40-R41